presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java

```diff
@@ -153,19 +153,19 @@ public void completionEvent(QueryInfo queryInfo)
         }
     }
 
-    private static Optional<TaskInfo> findFailedTask(StageInfo stageInfo)
+    private static Optional<TaskInfo> findFailedTask(Optional<StageInfo> stageInfo)
```

Comment:
Instead of making this take an Optional (which is a bit unnatural), make it take a StageInfo and call this method above like this:

```java
 Optional<TaskInfo> task = queryInfo.getOutputStage().map(QueryMonitor::findFailedTask);
```